internal/nostr/nostr.go

```diff
@@ -338,6 +339,40 @@ func (svc *Service) SubscriptionHandler(c echo.Context) error {
 	})
 }
 
+func (svc *Service) startSubscription(ctx context.Context, subscription *Subscription, sub *nostr.Subscription) error {
+	go func(){
+		for event := range sub.Events {
+			svc.Logger.WithFields(logrus.Fields{
+				"eventId":   event.ID,
+				"eventKind": event.Kind,
+			}).Infof("received event on subscription %d", subscription.ID)
+			responseEvent := ResponseEvent{
+				SubscriptionId: subscription.ID,
+				NostrId: event.ID,
+				Content: event.Content,
+				RepliedAt: event.CreatedAt.Time(),
+			}
+			svc.db.Save(&responseEvent)
+			svc.postEventToWebhook(event, subscription.WebhookUrl)
+		}
+	}()
+
+	select {
+	case <-sub.Relay.Context().Done():
+		svc.Logger.Errorf("relay error %v", sub.Relay.ConnectionError)
+		return sub.Relay.ConnectionError
+	case <-ctx.Done():
+		if ctx.Err() != context.Canceled {
+			svc.Logger.Errorf("subscription error %v", ctx.Err())
+			return ctx.Err()
+		}
+		svc.Logger.Infof("stopping subscription %d", subscription.ID)
+		svc.stopSubscription(subscription)
```

Comment:
This is called on shutdown, so it *would* mark all subscriptions as open=false (which means they would not start again on next startup), but there seems to be a bug there so that doesn't happen.

Reply:
Fixed!